tests/unit/test_asyn.py

```diff
@@ -111,6 +112,37 @@ async def process(row):
         list(mapper.iterate(timeout=4))
 
 
+@pytest.mark.parametrize("create_mapper", [AsyncMapper, OrderedMapper])
+def test_mapper_deadlock(create_mapper):
```

Comment:
could you elaborate a bit please? seems to be a regular mapper run, or is there something special about this config that makes it eligible to a deadlock, was there some bug in this configuration?